requirements.in

```diff
@@ -8,7 +12,10 @@ conllu
 
 # Candidate generation and entity linking
 joblib
-nmslib>=1.7.3.6
+nmslib>=1.7.3.6; python_version < '3.11'
+# Use the metabrainz fork until nmslib supports installing on Python 3.11+
+# https://github.com/nmslib/nmslib/issues/555
+nmslib-metabrainz>=2.1.3; python_version >= '3.11'
```

Comment:
I'm a bit hesitant about this because the metabrainz repo explicitly says they are not taking over ownership of nmslib. I wonder if it would be better to make nmslib an optional dependency, which would make it easier for one to install whatever version of nmslib works for them. This would be backwards incompatible as nmslib would no longer get installed by default, but maybe worth it.

Reply:
I was actually just able to install nmslib on linux with mamba and python 3.12.
```
mamba create -n py312 python=3.12
mamba install nmslib
```

Does that work for you?

Reply:
Unfortunately no (M1 Mac, tried with Micromamba and conda-forge channel), and our build system uses vanilla pip, so I don't think a conda package helps us.  That is interesting that someone got it to build on 3.12, though...

Reply:
fair enough. Lets pin nmslib-metabrainz exactly then? Not sure what to expect from that package going forward.

Reply:
Sure, works for me.